packages/eslint-plugin/tests/rules/no-enum-literals.test.ts

```diff
@@ -0,0 +1,142 @@
+import rule from '../../src/rules/no-enum-literals';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-enum-literals', rule, {
+  valid: [
+    '0 === 0',
+    'var a = 1',
+    'var b = "str"',
+    'var a = 1; (a === 0)',
+    'var a = 1; (0 === a)',
+    'var a = 1; (a === a)',
+    'var a = "str"; (a === "other")',
+    'var a: number = 1',
+    'var a: string = "str"',
+    'enum Foo { ONE, TWO }; let f: Foo; (f === Foo.ONE);',
+    'enum Foo { ONE, TWO }; let f: Foo; (Foo.ONE === f);',
+    'enum Foo { ONE, TWO }; let f = Foo.ONE;',
+    'enum Foo { ONE, TWO }; let f: Foo; f = Foo.ONE;',
+    'enum Foo { ONE, TWO }; function foo(f: Foo) {}; foo(Foo.ONE);',
+    'enum Foo { ONE = 1, TWO = 2 }; let f: Foo; (f === Foo.ONE);',
+    'enum Foo { ONE = 1, TWO = 2 }; let f: Foo; (Foo.ONE === f);',
+    'enum Foo { ONE = 1, TWO = 2 }; let f: Foo; f = Foo.ONE;',
+    'enum Foo { ONE = 1, TWO = 2 }; let f: Foo; f = Foo.ONE;',
+    'enum Foo { ONE = 1, TWO = 2 }; function foo(f: Foo) {}; foo(Foo.ONE);',
+    'enum Foo { ONE = "ONE", TWO = "TWO" }; let f: Foo; (f === Foo.ONE);',
+    'enum Foo { ONE = "ONE", TWO = "TWO" }; let f: Foo; (Foo.ONE === f);',
+    'enum Foo { ONE = "ONE", TWO = "TWO" }; let f: Foo; f = Foo.ONE;',
+    'enum Foo { ONE = "ONE", TWO = "TWO" }; let f: Foo = Foo.ONE;',
+    'enum Foo { ONE = "ONE", TWO = "TWO" }; function foo(f: Foo) {}; foo(Foo.ONE);',
+  ],
+
+  invalid: [
+    {
+      code: 'enum Foo { ONE, TWO }; let f: Foo; (f === 0);',
```

Comment:
please avoid one line test cases like this (and all of your invalid cases)
it's really hard to make out what failure it's testing because you only have the column number.